### ISSUE TITLE
JavadocVariable check wrongly reports that local variables in lambda expressions should be documented when the lambda is placed on a field initializer

### ISSUE THREAD

**Issue by miyakawataku** [checkstyle/checkstyle#1254]
When a lambda expression is placed on a field initializer, and the lambda body contains local variables, the local variables are wrongly detected by JavadocVariable check.

For example:

``` java
public class LambdaVar {
    /** Runnable. */
    private Runnable r = () -> {
        String str = "Hello world";
    };
}
```

JavadocVariable reports that `str` should be documented:

```
Starting audit...
/home/taku/pj/lambdavar/LambdaVar.java:4:9: Missing a Javadoc comment.
Audit done.
Checkstyle ends with 1 errors.
```
##

**Comment by ARoundSquare** [comment]
I've got a PR ready for this, just waiting on the JDK 8 upgrade (https://github.com/checkstyle/checkstyle/pull/3231) to get merged into master before I put mine up since the test class requires JDK 8.

**Comment by Vladlis** [comment]
@ARoundSquare 
> test class requires JDK 8

if you mean an input class, then such a resource can be placed at resources-noncompilable folder and marked as "compilable withi Java 8". Example: https://github.com/checkstyle/checkstyle/blob/master/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/InputStaticModifierInInterface.java

**Comment by romani** [comment]
@ARoundSquare, I am agree with Vladlis.

**Comment by ARoundSquare** [comment]
@Vladlis Ah, my mistake. I was confused by IntelliJ complaining about the language level of the resource folder. In that case, I'll have a PR out sometime this weekend.

**Comment by romani** [comment]
@ARoundSquare , please review http://checkstyle.sourceforge.net/idea.html

**Comment by ARoundSquare** [comment]
@romani , thanks!

**Comment by romani** [comment]
fix is merged.
